dot/rpc/modules/sync_state_test.go

```diff
@@ -1 +1,200 @@
+	type fields struct {
+		syncStateAPI SyncStateAPI
+	}
+	type args struct {
+		in0 *http.Request
+		req *GenSyncSpecRequest
+		res *genesis.Genesis
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+		err     error
+		exp     genesis.Genesis
+	}{
+		{
+			name: "GenSyncSpec OK",
+			fields: fields{
+				syncStateModule.syncStateAPI,
+			},
+			args: args{
+				req: &GenSyncSpecRequest{
+					Raw: true,
+				},
+			},
+			exp: genesis.Genesis{},
+		},
+		{
+			name: "GenSyncSpec Err",
+			fields: fields{
+				mockSyncStateAPIErr,
+			},
+			args: args{
+				req: &GenSyncSpecRequest{
+					Raw: true,
+				},
+			},
+			wantErr: true,
+			err:     errors.New("GenSyncSpec error"),
+		},
+	}
+	for _, tt := range tests {
+		var res genesis.Genesis
+		tt.args.res = &res
+		t.Run(tt.name, func(t *testing.T) {
+			ss := &SyncStateModule{
+				syncStateAPI: tt.fields.syncStateAPI,
+			}
+			var err error
+			if err = ss.GenSyncSpec(tt.args.in0, tt.args.req, tt.args.res); (err != nil) != tt.wantErr {
+				t.Errorf("GenSyncSpec() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if tt.wantErr {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.exp, *tt.args.res)
+			}
+		})
+	}
+}
+
+func TestNewStateSync(t *testing.T) {
+	g := &genesis.Genesis{}
+	raw := make(map[string][]byte)
+	mockStorageAPI := new(apimocks.StorageAPI)
+	mockStorageAPI.On("Entries", (*common.Hash)(nil)).Return(raw, nil)
+
+	mockStorageAPIErr := new(apimocks.StorageAPI)
+	mockStorageAPIErr.On("Entries", (*common.Hash)(nil)).Return(nil, errors.New("entries error"))
+
+	type args struct {
+		gData      *genesis.Data
+		storageAPI StorageAPI
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+		err     error
+		exp     SyncStateAPI
+	}{
+		{
+			name: "OK Case",
+			args: args{
+				gData:      g.GenesisData(),
+				storageAPI: mockStorageAPI,
+			},
+			exp: syncState{chainSpecification: &genesis.Genesis{
+				Name:       "",
+				ID:         "",
+				Bootnodes:  []string{},
+				ProtocolID: "",
+				Genesis: genesis.Fields{
+					Raw:     map[string]map[string]string{},
+					Runtime: map[string]map[string]interface{}{},
+				},
+			},
+			},
+		},
+		{
+			name: "Err Case",
+			args: args{
+				gData:      g.GenesisData(),
+				storageAPI: mockStorageAPIErr,
+			},
+			wantErr: true,
+			err:     errors.New("entries error"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			res, err := NewStateSync(tt.args.gData, tt.args.storageAPI)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewStateSync() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if tt.wantErr {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.exp, res)
+			}
+		})
+	}
+}
+
+func Test_syncState_GenSyncSpec(t *testing.T) {
+	g := &genesis.Genesis{}
```

Comment:
Define inline in your test cases

i.e.

```go
fields: fields{
    chainSpecification: &genesis.Genesis{},
},
```

For readability and to avoid subtests to eventually depend on each other

Reply:
Good call. The other instance of this in the above function I wasn't able to inline, but there were only 2 instances so I just made a second variable for it